src/keycard/keycard.cljs

```diff
@@ -147,17 +148,23 @@
 
 (defn generate-and-load-key
   [{:keys [mnemonic pin on-success on-failure]}]
-  (.. status-keycard
-      (generateAndLoadKey mnemonic pin)
-      (then on-success)
-      (catch on-failure)))
+  (native-module/validate-mnemonic
```

Comment:
sorry i know i suggested that, but it's better to move validation outside, keycard ns should know about native-module

Reply:
I have moved validation outside, but [out of 4](https://github.com/status-im/status-mobile/issues/22209#issuecomment-2703826927), only 2 functions are being used. So I can only apply validation there. Please let me know, it looks ok?